gulpfile.js

```diff
@@ -68,6 +70,39 @@ gulp.task('watch', function () {
   }
 });
 
+// Watch Server files for changes
+gulp.task('watch_server_files_for_testing', function () {
+  var runOnlyChangedTestFile = argv.onlyChanged ? true : false;
+
+  // Start livereload
+  plugins.livereload.listen();
+
+  // Add Server Test file rules
+  gulp.watch([testAssets.tests.server, defaultAssets.server.allJS], ['test:server']).on('change', function (file) {
+    // check if we should only run a changed test file
+    if (runOnlyChangedTestFile) {
+      var changedTestFiles = [];
+
+      // iterate through server test glob patterns
+      _.forEach(testAssets.tests.server, function (tests) {
+        // determine if the changed (watched) file is a server test
+        _.forEach(glob.sync(tests), function (f) {
```

Comment:
Not really sure if this is a glob thing, but why not just check if the intersection of the two lists (watched files and assets.tests.server) is nonzero?